providers/loading.tsx

```diff
@@ -0,0 +1,19 @@
+import { createContext, Dispatch, SetStateAction, useState } from "react";
+import Loading from "../components/loading";
+
+let setStateAction: Dispatch<SetStateAction<boolean>>;
+
+const LoadingContext = createContext(setStateAction);
+
+function LoadingContextProvider({ children }) {
+  const [loading, setLoading] = useState<boolean>(false);
```

Comment:
make this be global to this file, so we can export `loading` variable in case we might need to read the state,

```ts
// ...
const LoadingContext = createContext(setStateAction);
const [isLoading, setLoading] = useState<boolean>(false);

function LoadingContextProvider...
```

Reply:
done

Reply:
![image](https://user-images.githubusercontent.com/34746557/124309217-c6861d00-db40-11eb-9f02-b62a9ce9032f.png)

Reply:
yeah, I should have tested that before :D

lets go the cheeky route:

```ts
//...
let getLoadingState: () => boolean;
// ...
function LoadingContextProvider {
    const [loading, setLoading] = useState<boolean>(false);
    getLoadingState = () => loading;
    // ...
}

export { LoadingContextProvider, LoadingContext, getLoadingState };

Reply:
Good, i adjusted. Done ^.^

Reply:
> use state only works within a component

You always call hooks within CamelCase components or modules, as helpers itself. Example:
```
function Component() {
   const [S, D] = useState()
}
```

and not as
```
function component() {
   const [S, D] = useState()
}
```